src/codec.rs

```diff
@@ -1093,7 +1094,7 @@ pub(crate) fn compact_encode_len_to<W: Output + ?Sized>(dest: &mut W, len: usize
 
 impl<T: Encode> Encode for [T] {
 	fn size_hint(&self) -> usize {
-		mem::size_of::<u32>() + mem::size_of::<T>() * self.len()
+		mem::size_of::<u32>() + core::mem::size_of_val(self)
```

Comment:
Not sure if saturating add would make sense here.

Reply:
Hmm... perhaps, but then it's just a size hint so shouldn't really matter if it overflows.

Reply:
I dont know if the runtime is normally compiled with overflow checks enabled (that would be my only concern).

Reply:
Overflow checks are not enabled for release builds as they would be too slow.